app/screens/buyers/BuyingItemList.java

```diff
@@ -1,44 +1,90 @@
+import java.util.ArrayList;
+import java.util.List;
 import java.util.Scanner;
 
 import icebase.app.App;
 import icebase.app.MenuTitle;
+import icebase.app.api.API;
+import icebase.app.enums.CATEGORY_ENUM;
+import icebase.app.exceptions.CategoryDoesNotExistException;
 import icebase.app.screens.Screen;
-import java.util.Scanner;
+import icebase.app.types.Store;
+import icebase.app.types.items.Item;
+import icebase.icebase.Auth;
+import icebase.icebase.User;
 
 public class BuyingItemList implements Screen {
 
-    public void display() {
-        Scanner sc = App.sc;
-        int choice;
-        String storeName = "STORE X"; // stand-in value, should be set from constructor
-        String categoryName = "CATEGORY X"; // stand-in value, should be set from constructor
-        String[] itemList = { "Item A", "Item B", "Item C", "Return" }; // fetch from icebase
-        int range = itemList.length;
+    private String storeName;
+    private String categoryName;
+    private List<Item> itemList;
+    private ArrayList<String> itemNames;
+    private Scanner sc = App.sc;
+    private Auth auth = Auth.getAuth();
+    private User user = auth.getUser();
+
+    public BuyingItemList(Store store, CATEGORY_ENUM category) {
+        this.storeName = store.getName();
+        this.categoryName = category.value;
+        try {
+            this.itemList = API.getStoreItems(store, category);
+            for (Item item : itemList) {
+                itemNames.add(item.getName());
+            }
+            itemNames.add("return");
+
+        } catch (IOException io) {
+            io.printStackTrace();
+        } catch (CategoryDoesNotExistException cdne) {
+            cdne.printStackTrace();
+        }
+
+    }
 
+    public void display() {
         while (true) {
+            int choice;
             MenuTitle.displayStoreName(storeName);
-            MenuTitle.displaySubTitle("[" + categoryName + "]");
-            // MenuTitle.displayOptions(itemList);
+            MenuTitle.displaySubTitle(categoryName);
+            MenuTitle.displayOptions(itemNames);
 
             System.out.print("\nChoice: ");
             try {
                 choice = Integer.parseInt(sc.nextLine());
-                if (choice == range + 1) {
-                    break;
+                if (choice == itemList.size()) {
+                    return;
                 }
-                String item = itemList[choice - 1]; // fetch from icebase
-                buyItem(item);
+                Item item = itemList.get(choice - 1);
+                this.buyItem(item);
             } catch (Exception e) {
                 System.out.println("Please choose from the given options...\n");
             }
         }
     }
 
-    public static void buyItem(String item) {
-        System.out.println(item);
-        // display other item attributes
+    public void buyItem(Item item) {
+        int amount;
+        String itemName = item.getName();
+        while (true) {
```

Comment:
why naa pay while true? i dont think ga loop ta here?

Reply:
kay pwede man nga invalid amount i enter, string and stuff, pero pwede ra nga ang prompt nalang ienclose sa while

Reply:
ah okay kay related sa amount nga code mo, i think better nga enclosed ang while kay di siya nice tingnan ang throw Exception